src/csharp/Microsoft.Spark/Sql/ForeachWriter.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Spark.Sql
+{
+    /// <summary>
+    /// Interface for writing custom logic to process data generated by a query. This is
+    /// often used to write the output of a streaming query to arbitrary storage systems.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Any implementation of this interface will be used by Spark in the following way:
+    /// <list type="bullet">
+    /// <item>
+    /// <description>
+    /// A single instance of this class is responsible of all the data generated by a single task
```

Comment:
```suggestion
    /// A single instance of this class is responsible for all the data generated by a single task
```

Reply:
At the moment this matches the spark description.  I'll leave this as is.